src/client_side.cc

```diff
@@ -1636,7 +1636,7 @@ clientProcessRequest(ConnStateData *conn, const Http1::RequestParserPointer &hp,
 #endif
 
     if (internalCheck(request->url.path())) {
-        if (internalHostnameIs(request->url.host()) && request->url.port() == getMyPort()) {
+        if (internalWithMyHostname(request->url, *(conn->port))) {
```

Comment:
FWIW, I see no need to add those parenthesis to dereference a pointer: If there is a way for the alternative `(*conn)->port()` expression to compile, then we have a much bigger problem that cannot be solved by sprinkling parenthesis around the code! We can and should keep it simple:

```suggestion
        if (internalWithMyHostname(request->url, *conn->port)) {
```

Reply:
These brackets are for developer readability, to clarity that the `port` pointer is being dereferenced not the `conn` pointer.